sympy/core/numbers.py

```diff
@@ -444,12 +426,10 @@ def ilcm(*args):
     if len(args) < 2:
         raise TypeError(
             'ilcm() takes at least 2 arguments (%s given)' % len(args))
-    if 0 in args:
-        return 0
-    a = args[0]
-    for b in args[1:]:
-        a = a // igcd(a, b) * b # since gcd(a,b) | a
-    return a
+    ret = int(number_lcm(*map(as_int, args)))
+    if issubclass(type(args[0]), Integer):
```

Comment:
This faithfully retains the legacy behavior, but it should really be returning an int.

Reply:
You are right. However, as you can see from the failed test of the first commit, there is a function that expects Integer to be returned. I should probably fix the function that is calling it, but I don't think I can handle it.

Reply:
In the `prde` file three lines can be changed:

```python
n = reduce(ilcm, [i.as_numer_denom()[1] for i in u])
n = reduce(ilcm, [i.as_numer_denom()[1] for _, i in residueterms], S.One)
common_denom = reduce(ilcm, [i.as_numer_denom()[1] for i in [j for _, j in
        residueterms]] + [n], S.One)
```
to
```python
n = S.One*reduce(ilcm, [i.as_numer_denom()[1] for i in u])
n = S.One*reduce(ilcm, [i.as_numer_denom()[1] for _, i in residueterms], 1)
common_denom = S.One*reduce(ilcm, [i.as_numer_denom()[1] for i in [j for _, j in
        residueterms]] + [n], 1)
```

Note that all get multiplied by S.One and the last two lines use a default of 1 instead of S.One

Reply:
Thank you.